src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,284 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+from  util import winnforum_testcase, makePpaAndPalRecordsConsistent,\
+ configurable_testcase, writeConfig, loadConfig
+import sas_testcase
+import hashlib
+from datetime import datetime, timedelta
+
+class FullActivityDumpMessageTestcase(sas_testcase.SasTestCase):
+
+    def setUp(self):
+        self._sas, self._sas_admin = sas.GetTestingSas()
+        self._sas_admin.Reset()
+
+    def tearDown(self):
+        pass
+    
+    def generate_FAD_1_default_config(self, filename):
```

Comment:
Prefer to put this function directly above the FAD_1 testcase itself so that it's easy to keep them in sync.